To: vim_dev@googlegroups.com
Subject: Patch 7.3.191
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.191
Problem:    Still some RISC OS stuff to remove.
Solution:   Remove files and lines. (Hong Xu)
	    Remove the 'osfiletype' option code.
Files:	    README_extra.txt, src/Make_ro.mak, src/INSTALL, src/Makefile,
	    src/buffer.c, src/eval.c, src/feature.h, src/option.c,
	    src/option.h, src/structs.h, src/version.c, src/pty.c, Filelist


*** ../vim-7.3.190/README_extra.txt	2010-08-15 21:57:32.000000000 +0200
--- README_extra.txt	2011-05-19 12:35:16.000000000 +0200
***************
*** 13,21 ****
  
  src/os_amiga.*		Files for the Amiga port.
  
- src/gui_riscos.*
- src/os_riscos.*		Files for the RISC OS port.
- 
  src/gui_beos.*
  src/os_beos.*		Files for the BeOS port.
  
--- 13,18 ----
*** ../vim-7.3.190/src/Make_ro.mak	2010-08-15 21:57:27.000000000 +0200
--- src/Make_ro.mak	1970-01-01 01:00:00.000000000 +0100
***************
*** 1,135 ****
- #
- # Makefile for Vim on RISC OS - Andy Wingate
- #
- 
- GCC         = gcc -mthrowback
- CFLAGS     = -DRISCOS -DFEAT_GUI
- CC         = $(GCC) $(CFLAGS) -O2
- # -DUP_BC_PC_EXTERN for term.c needed as BC defined in termlib.c and term.c
- 
- TERMFLAG   = -DUP_BC_PC_EXTERN
- 
- ASMFLAGS   = -throwback -objasm -gcc
- 
- OBJS =  o.buffer o.charset o.diff o.digraph o.edit o.eval o.ex_cmds o.ex_cmds2  \
- 	o.ex_docmd o.ex_eval o.ex_getln o.fileio o.fold o.getchar \
- 	o.hardcopy o.hashtab o.main o.mark o.mbyte  \
- 	o.memfile o.memline o.menu o.message o.misc1 o.misc2 o.move     \
- 	o.normal o.ops o.option o.popupmnu o.quickfix o.regexp o.screen \
- 	o.search   \
- 	o.spell o.syntax o.tag o.term o.termlib o.ui o.undo o.version	\
- 	o.window o.os_riscos o.swis o.gui o.gui_riscos
- 
- Vim: $(OBJS)
- 	$(GCC) -o Vim $(OBJS)
- 
- install: Vim
- 	squeeze -v Vim @.!Vim.Vim
- 
- clean:	
- 	create o.!fake! 0
- 	wipe o.* ~cf
- 	remove Vim
- 
- o.swis: s.swis
- 	as $(ASMFLAGS) -o o.swis s.swis
- 
- # Rules for object files
- 
- o.%:	c.%
- 	$(CC) -c $< -o $@
- 
- o.buffer:	c.buffer
- 
- o.charset:	c.charset
- 
- o.digraph:	c.digraph
- 
- o.diff:		c.diff
- 
- o.edit:		c.edit
- 
- o.eval:		c.eval
- 
- o.ex_cmds:	c.ex_cmds
- 
- o.ex_cmds2:	c.ex_cmds2
- 
- o.ex_docmd:	c.ex_docmd
- 
- o.ex_eval:	c.ex_eval
- 
- o.ex_getln:	c.ex_getln
- 
- o.fileio:	c.fileio
- 
- o.fold:		c.fold
- 
- o.getchar:	c.getchar
- 
- o.hardcopy:	c.hardcopy
- 
- o.hashtab:	c.hashtab
- 
- o.gui:		c.gui
- 
- o.gui_riscos:	c.gui_riscos
- 
- o.main:		c.main
- 
- o.mark:		c.mark
- 
- o.mbyte:	c.mbyte
- 
- o.memfile:	c.memfile
- 
- o.memline:	c.memline
- 
- o.menu:		c.menu
- 
- o.message:	c.message
- 
- o.misc1:	c.misc1
- 
- o.misc2:	c.misc2
- 
- o.move:		c.move
- 
- o.normal:	c.normal
- 
- o.ops:		c.ops
- 
- o.option:	c.option
- 
- o.os_riscos:	c.os_riscos
- 
- o.pty:		c.pty
- 
- o.popupmnu:	c.popupmnu
- 
- o.quickfix:	c.quickfix
- 
- o.regexp:	c.regexp
- 
- o.screen:	c.screen
- 
- o.search:	c.search
- 
- o.spell:	c.spell
- 
- o.syntax:	c.syntax
- 
- o.tag:		c.tag
- 
- o.term:		c.term
- 	$(CC) $(TERMFLAG) -c c.term -o o.term
- 
- o.termlib:	c.termlib
- 
- o.ui:		c.ui
- 
- o.undo:		c.undo
- 
- o.version:	c.version
- 
- o.window:	c.window
--- 0 ----
*** ../vim-7.3.190/src/INSTALL	2010-08-15 21:57:28.000000000 +0200
--- src/INSTALL	2011-05-19 12:36:17.000000000 +0200
***************
*** 6,14 ****
  Contents:
  1. Generic
  2. Unix
! 3. RISC OS
! 4. OS/2 (with EMX 0.9b)
! 5. Atari MiNT
  
  See INSTALLami.txt              for Amiga
  See INSTALLmac.txt              for Macintosh
--- 6,13 ----
  Contents:
  1. Generic
  2. Unix
! 3. OS/2 (with EMX 0.9b)
! 4. Atari MiNT
  
  See INSTALLami.txt              for Amiga
  See INSTALLmac.txt              for Macintosh
***************
*** 174,198 ****
  	./configure --without-local-dir
  
  
! 3. RISC OS
! =============
! 
! Much file renaming is needed before you can compile anything.
! You'll need UnixLib to link against, GCC and GNU make.
! 
! I suggest you get the RISC OS binary distribution, which includes the
! Templates file and the loader.
! 
! Try here: http://www.ecs.soton.ac.uk/~tal197
! 
! Do
!     :help riscos
! 
! within the editor for more information, or read the
! ../runtime/doc/os_risc.txt help file.
! 
! 
! 4. OS/2
  =======
  
  Summary:
--- 173,179 ----
  	./configure --without-local-dir
  
  
! 3. OS/2
  =======
  
  Summary:
***************
*** 237,243 ****
  Check ../runtime/doc/os_os2.txt for additional info on running Vim.
  
  
! 5. Atari MiNT
  =============
  
  [NOTE: this is quite old, it might not work anymore]
--- 218,224 ----
  Check ../runtime/doc/os_os2.txt for additional info on running Vim.
  
  
! 4. Atari MiNT
  =============
  
  [NOTE: this is quite old, it might not work anymore]
*** ../vim-7.3.190/src/Makefile	2011-04-01 13:05:37.000000000 +0200
--- src/Makefile	2011-05-19 12:37:52.000000000 +0200
***************
*** 29,35 ****
  #	- Uncomment the line "CONF_OPT_X = --without-x" if you have X11 but
  #	  want to disable using X11 libraries.	This speeds up starting Vim,
  #	  but the window title will not be set and the X11 selection can not
! #	  used.
  #	- Uncomment the line "CONF_OPT_XSMP = --disable-xsmp" if you have the
  #	  X11 Session Management Protocol (XSMP) library (libSM) but do not
  #	  want to use it.
--- 29,35 ----
  #	- Uncomment the line "CONF_OPT_X = --without-x" if you have X11 but
  #	  want to disable using X11 libraries.	This speeds up starting Vim,
  #	  but the window title will not be set and the X11 selection can not
! #	  be used.
  #	- Uncomment the line "CONF_OPT_XSMP = --disable-xsmp" if you have the
  #	  X11 Session Management Protocol (XSMP) library (libSM) but do not
  #	  want to use it.
***************
*** 730,741 ****
  ### USL for Unix Systems Laboratories (SYSV 4.2)
  #EXTRA_DEFS = -DUSL
  
- ### RISCos on MIPS without X11
- #EXTRA_DEFS = -DMIPS
- 
- ### RISCos on MIPS with X11
- #EXTRA_LIBS = -lsun
- 
  ### (6)  A/UX 3.1.1 with gcc (Jim Jagielski)
  #CC= gcc -D_POSIX_SOURCE
  #CFLAGS= -O2
--- 730,735 ----
***************
*** 1633,1639 ****
  RSRC_DIR = os_mac_rsrc
  
  PRO_MANUAL = os_amiga.pro os_msdos.pro os_win16.pro os_win32.pro \
! 	os_mswin.pro os_beos.pro os_vms.pro os_riscos.pro $(PERL_PRO)
  
  # Default target is making the executable and tools
  all: $(VIMTARGET) $(TOOLS) languages $(GUI_BUNDLE)
--- 1627,1633 ----
  RSRC_DIR = os_mac_rsrc
  
  PRO_MANUAL = os_amiga.pro os_msdos.pro os_win16.pro os_win32.pro \
! 	os_mswin.pro os_beos.pro os_vms.pro $(PERL_PRO)
  
  # Default target is making the executable and tools
  all: $(VIMTARGET) $(TOOLS) languages $(GUI_BUNDLE)
*** ../vim-7.3.190/src/buffer.c	2011-05-05 17:32:40.000000000 +0200
--- src/buffer.c	2011-05-19 13:09:25.000000000 +0200
***************
*** 1808,1816 ****
  #ifdef FEAT_AUTOCMD
      clear_string_option(&buf->b_p_ft);
  #endif
- #ifdef FEAT_OSFILETYPE
-     clear_string_option(&buf->b_p_oft);
- #endif
  #ifdef FEAT_CINDENT
      clear_string_option(&buf->b_p_cink);
      clear_string_option(&buf->b_p_cino);
--- 1808,1813 ----
*** ../vim-7.3.190/src/eval.c	2011-05-10 16:41:13.000000000 +0200
--- src/eval.c	2011-05-19 13:09:43.000000000 +0200
***************
*** 12076,12084 ****
  #ifdef FEAT_OLE
  	"ole",
  #endif
- #ifdef FEAT_OSFILETYPE
- 	"osfiletype",
- #endif
  #ifdef FEAT_PATH_EXTRA
  	"path_extra",
  #endif
--- 12076,12081 ----
*** ../vim-7.3.190/src/feature.h	2010-08-15 21:57:31.000000000 +0200
--- src/feature.h	2011-05-19 13:09:56.000000000 +0200
***************
*** 506,520 ****
  #endif
  
  /*
-  * +osfiletype		filetype checking in autocommand patterns.
-  *			Only on systems that support filetypes (RISC OS).
-  */
- #if 0
- # define FEAT_OSFILETYPE
- # define DFLT_OFT "Text"
- #endif
- 
- /*
   * +viminfo		reading/writing the viminfo file. Takes about 8Kbyte
   *			of code.
   * VIMINFO_FILE		Location of user .viminfo file (should start with $).
--- 506,511 ----
*** ../vim-7.3.190/src/option.c	2011-04-28 17:24:54.000000000 +0200
--- src/option.c	2011-05-19 13:11:42.000000000 +0200
***************
*** 140,148 ****
  #define PV_MOD		OPT_BUF(BV_MOD)
  #define PV_MPS		OPT_BUF(BV_MPS)
  #define PV_NF		OPT_BUF(BV_NF)
- #ifdef FEAT_OSFILETYPE
- # define PV_OFT		OPT_BUF(BV_OFT)
- #endif
  #ifdef FEAT_COMPL_FUNC
  # define PV_OFU		OPT_BUF(BV_OFU)
  #endif
--- 140,145 ----
***************
*** 337,345 ****
  static int	p_mod;
  static char_u	*p_mps;
  static char_u	*p_nf;
- #ifdef FEAT_OSFILETYPE
- static char_u	*p_oft;
- #endif
  static int	p_pi;
  #ifdef FEAT_TEXTOBJ
  static char_u	*p_qe;
--- 334,339 ----
***************
*** 1901,1914 ****
  			    (char_u *)NULL, PV_NONE,
  			    {(char_u *)FALSE, (char_u *)0L} SCRIPTID_INIT},
      {"osfiletype",  "oft",  P_STRING|P_ALLOCED|P_VI_DEF,
- #ifdef FEAT_OSFILETYPE
- 			    (char_u *)&p_oft, PV_OFT,
- 			    {(char_u *)DFLT_OFT, (char_u *)0L}
- #else
  			    (char_u *)NULL, PV_NONE,
! 			    {(char_u *)0L, (char_u *)0L}
! #endif
! 			    SCRIPTID_INIT},
      {"paragraphs",  "para", P_STRING|P_VI_DEF,
  			    (char_u *)&p_para, PV_NONE,
  			    {(char_u *)"IPLPPPQPP TPHPLIPpLpItpplpipbp",
--- 1895,1902 ----
  			    (char_u *)NULL, PV_NONE,
  			    {(char_u *)FALSE, (char_u *)0L} SCRIPTID_INIT},
      {"osfiletype",  "oft",  P_STRING|P_ALLOCED|P_VI_DEF,
  			    (char_u *)NULL, PV_NONE,
! 			    {(char_u *)0L, (char_u *)0L} SCRIPTID_INIT},
      {"paragraphs",  "para", P_STRING|P_VI_DEF,
  			    (char_u *)&p_para, PV_NONE,
  			    {(char_u *)"IPLPPPQPP TPHPLIPpLpItpplpipbp",
***************
*** 5282,5290 ****
  #ifdef FEAT_AUTOCMD
      check_string_option(&buf->b_p_ft);
  #endif
- #ifdef FEAT_OSFILETYPE
-     check_string_option(&buf->b_p_oft);
- #endif
  #if defined(FEAT_SMARTINDENT) || defined(FEAT_CINDENT)
      check_string_option(&buf->b_p_cinw);
  #endif
--- 5270,5275 ----
***************
*** 9665,9673 ****
  	case PV_MA:	return (char_u *)&(curbuf->b_p_ma);
  	case PV_MOD:	return (char_u *)&(curbuf->b_changed);
  	case PV_NF:	return (char_u *)&(curbuf->b_p_nf);
- #ifdef FEAT_OSFILETYPE
- 	case PV_OFT:	return (char_u *)&(curbuf->b_p_oft);
- #endif
  	case PV_PI:	return (char_u *)&(curbuf->b_p_pi);
  #ifdef FEAT_TEXTOBJ
  	case PV_QE:	return (char_u *)&(curbuf->b_p_qe);
--- 9650,9655 ----
***************
*** 10018,10026 ****
  	    /* Don't copy 'filetype', it must be detected */
  	    buf->b_p_ft = empty_option;
  #endif
- #ifdef FEAT_OSFILETYPE
- 	    buf->b_p_oft = vim_strsave(p_oft);
- #endif
  	    buf->b_p_pi = p_pi;
  #if defined(FEAT_SMARTINDENT) || defined(FEAT_CINDENT)
  	    buf->b_p_cinw = vim_strsave(p_cinw);
--- 10000,10005 ----
*** ../vim-7.3.190/src/option.h	2011-02-12 13:59:55.000000000 +0100
--- src/option.h	2011-05-19 13:11:46.000000000 +0200
***************
*** 984,992 ****
      , BV_MOD
      , BV_MPS
      , BV_NF
- #ifdef FEAT_OSFILETYPE
-     , BV_OFT
- #endif
  #ifdef FEAT_COMPL_FUNC
      , BV_OFU
  #endif
--- 984,989 ----
*** ../vim-7.3.190/src/structs.h	2011-05-10 16:41:13.000000000 +0200
--- src/structs.h	2011-05-19 13:11:58.000000000 +0200
***************
*** 1530,1538 ****
      int		b_p_ml_nobin;	/* b_p_ml saved for binary mode */
      int		b_p_ma;		/* 'modifiable' */
      char_u	*b_p_nf;	/* 'nrformats' */
- #ifdef FEAT_OSFILETYPE
-     char_u	*b_p_oft;	/* 'osfiletype' */
- #endif
      int		b_p_pi;		/* 'preserveindent' */
  #ifdef FEAT_TEXTOBJ
      char_u	*b_p_qe;	/* 'quoteescape' */
--- 1530,1535 ----
*** ../vim-7.3.190/src/version.c	2011-05-19 12:14:03.000000000 +0200
--- src/version.c	2011-05-19 13:26:42.000000000 +0200
***************
*** 426,436 ****
  	"-ole",
  # endif
  #endif
- #ifdef FEAT_OSFILETYPE
- 	"+osfiletype",
- #else
- 	"-osfiletype",
- #endif
  #ifdef FEAT_PATH_EXTRA
  	"+path_extra",
  #else
--- 426,431 ----
*** ../vim-7.3.190/src/pty.c	2011-04-11 14:24:33.000000000 +0200
--- src/pty.c	2011-05-19 12:43:26.000000000 +0200
***************
*** 123,130 ****
  static void initmaster __ARGS((int));
  
  /*
!  *  Open all ptys with O_NOCTTY, just to be on the safe side
!  *  (RISCos mips breaks otherwise)
   */
  #ifndef O_NOCTTY
  # define O_NOCTTY 0
--- 123,129 ----
  static void initmaster __ARGS((int));
  
  /*
!  *  Open all ptys with O_NOCTTY, just to be on the safe side.
   */
  #ifndef O_NOCTTY
  # define O_NOCTTY 0
*** ../vim-7.3.190/Filelist	2011-05-10 17:21:34.000000000 +0200
--- Filelist	2011-05-19 13:37:25.000000000 +0200
***************
*** 427,433 ****
  		$(SRC_VMS) \
  		README_os390.txt \
  		src/Make_mint.mak \
- 		src/Make_ro.mak \
  		src/if_sniff.c \
  		src/infplist.xml \
  		src/link.390 \
--- 427,432 ----
*** ../vim-7.3.190/src/version.c	2011-05-19 12:14:03.000000000 +0200
--- src/version.c	2011-05-19 13:26:42.000000000 +0200
***************
*** 716,717 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     191,
  /**/

-- 
From "know your smileys":
 :-&	Eating spaghetti

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
